### DISCUSSION TITLE
How to connect a customdevice to a HASS? (exposes + HASS)

### DISCUSSION THREAD

**Discussion by SandrRu** [Koenkk/zigbee2mqtt#5605] [Devices]
Hello dear friends.

I cant get new custom device in HASS.

I write converter for PWM controller with ptvo firmware:
[sandr_ru.js] (https://github.com/Koenkk/zigbee2mqtt/files/5780365/sandr_ru.txt)

Exposes in z2m web ui work good

![2021-01-07_11-19-49](https://user-images.githubusercontent.com/77043365/103868712-4ffceb00-50da-11eb-9986-6ce95ee7e73f.png)
![2021-01-07_11-19-40](https://user-images.githubusercontent.com/77043365/103868716-50958180-50da-11eb-9692-abea64054346.png)

But I cant get this device In HASS device list....

Can anybody help me ?

**Comment by Koenkk** [comment]
Can you check if this is fixed in the latest dev branch?

Changes will be available in the latest dev branch in a few hours (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Reply by SandrRu** [reply]
Thank you for your work !
In Dev brach work good/
![2021-01-07_23-27-18](https://user-images.githubusercontent.com/77043365/103941501-00510a80-5140-11eb-9125-0018c72c2861.png)
